coordinator/coordinator_test.go

```diff
@@ -114,3 +116,36 @@ func TestServiceCoordinator(t *testing.T) {
 		t.Error("Stop was not called in the expected timeframe")
 	}
 }
+
+func exampleNewHTTPServer() *stubProcess {
+	return new(stubProcess)
+}
+func exampleNewPrometheusMetrics() *stubProcess {
+	return new(stubProcess)
+}
+
+func Example_newServiceCoordinator() {
+	// By assuming you need to wrap you application core startup logic in one handled algorithm.
+	// Things like how to start and stop server, monitoring, metrics in controllable goroutines.
+	// Then you can simplify your code base by delighting it to `ServiceCoordinator`.
+	//
+	// Before creating new ServiceCoordinator
+	// You may need to define configuration, like reading env vars or file
+	// Then you can create new instance of ServiceCoordinator
+	applicationForceStopTimeout := time.Minute
+	setOfOptions := []Options{
+		SetForceStopTimeout(applicationForceStopTimeout),
+		// each of processes must be implementation of type Process interface from `process` package.
+		AddProcesses(exampleNewHTTPServer(), exampleNewPrometheusMetrics()),
+	}
+
+	// Now pass your setup to service coordinator
+	sc := NewServiceCoordinator(setOfOptions...)
+
+	// And you can start it, inside OS signals will be handled, so you don't need to write graceful shutdown here.
+	if err := sc.Start(); err != nil {
```

Comment:
We should add a TODO somewhere to open up signals and allow a channel to be passed so that it can shutdown optionally.

Reply:
@alfredosa We can create issue in repository with code snippet /reference actually :)